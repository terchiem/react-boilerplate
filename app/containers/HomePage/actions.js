import {
  GET_POSTS,
  GET_POSTS_SUCCESS,
  GET_POSTS_FAIL,
  ADD_POST_TO_LIST,
} from './constants';

export function getPosts() {
  return {
    type: GET_POSTS,
  };
}

export function getPostsSuccess(posts) {
  return {
    type: GET_POSTS_SUCCESS,
    posts,
  };
}

export function getPostsFail(error) {
  return {
    type: GET_POSTS_FAIL,
    error,
  };
}

export function addPostToList(post) {
  return {
    type: ADD_POST_TO_LIST,
    post,
  };
}
