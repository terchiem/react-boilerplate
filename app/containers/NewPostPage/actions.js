import { CHANGE_POST_INPUT, RESET_POST_INPUT } from './constants';

export function changePostInput(value) {
  return {
    type: CHANGE_POST_INPUT,
    value,
  };
}

export function resetPostInput() {
  return {
    type: RESET_POST_INPUT,
  };
}
