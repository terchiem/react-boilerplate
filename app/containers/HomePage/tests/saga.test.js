/**
 * Tests for HomePage sagas
 */

import { put, takeLatest } from 'redux-saga/effects';

import { GET_POSTS } from '../constants';
import { getPostsSuccess, getPostsFail } from '../actions';

import getPostsSaga, { getPostList } from '../saga';

/* eslint-disable redux-saga/yield-effects */
describe('getPosts Saga', () => {
  let getPostListGenerator;

  beforeEach(() => {
    getPostListGenerator = getPostList();

    const callDescriptor = getPostListGenerator.next().value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('should dispatch the getPostsSuccess action if it requests the data successfully', () => {
    const response = {
      posts: ['Post 1', 'Post 2'],
    };
    const putDescriptor = getPostListGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(getPostsSuccess(response.posts)));
  });

  it('should call the getPostsFail action if the response errors', () => {
    const response = new Error('Some error');
    const putDescriptor = getPostListGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(getPostsFail(response)));
  });
});

describe('getPostsSaga Saga', () => {
  const saga = getPostsSaga();

  it('should start task to watch for LOAD_REPOS action', () => {
    const takeLatestDescriptor = saga.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(GET_POSTS, getPostList));
  });
});
