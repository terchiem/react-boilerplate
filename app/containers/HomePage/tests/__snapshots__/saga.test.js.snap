// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPosts Saga should call the getPostsFail action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "http://localhost:3000/api",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getPosts Saga should dispatch the getPostsSuccess action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "http://localhost:3000/api",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
